src/sentry/services/hybrid_cloud/user/impl.py

```diff
@@ -182,12 +186,25 @@ def get_or_create_user_by_email(self, *, email: str) -> RpcUser:
                     email=email,
                     name=email,
                 )
+                user_signup.send_robust(
+                    sender=self, user=user, source="api", referrer=referrer or "unknown"
+                )
             else:
                 # Users are not supposed to have the same email but right now our auth pipeline let this happen
-                # So let's not break the user experience
+                # So let's not break the user experience. Instead return the user with auth identity of ident or
+                # the first user if ident is None
+                user = user_query[0]
                 if user_query.count() > 1:
                     logger.warning("Email has multiple users", extra={"email": email})
-                user = user_query[0]
+                    if ident:
+                        for candid_user in user_query:
+                            identity_query = AuthIdentity.objects.filter(
+                                user=candid_user, ident=ident
+                            )
```

Comment:
Looks like an N+1 query? Probably want to do something like

```python
AuthIdentity.objects.filter(user__in=user_query, ident=ident)
```

and then loop over the result.

It might also be worth logging if that query returns multiple `AuthIdentity` objects.